crates/bevy_ui/src/geometry.rs

```diff
@@ -282,6 +282,54 @@ impl Val {
     }
 }
 
+/// Returns a [`Val::Auto`] where the value is automatically determined
+/// based on the context and other [`Node`](crate::Node) properties.
+pub const fn auto() -> Val {
+    Val::Auto
+}
+
+/// Returns a [`Val::Px`] representing a value in logical pixels.
+pub const fn px(value: f32) -> Val {
```

Comment:
The alternative to this is adding `use Val::*` to the prelude, which feels less "special" to me, ties the docs directly to `Val`, keeps our code size down, and aligns with Rust std patterns (ex: `Some`, `None`, `Ok`, `Err`, etc).

The only problem is right now `bsn!` breaks on this case (as it assumes it can do `parent_type.value.0` to set `Some(x)`). This is obviously a problem! The solution is to use `if let` to destructure all types. I believe this will also solve one of the outstanding autocomplete issues (and it will remove the need for the `touch_type` hack).

Reply:
Arg yeah that does make this comfier. And currently type inference actually doesn't break in `bsn!` right now, as function arguments don't have implicit `into()`.

Reply:
It will be done by BSN. There's automatic `.into()` calls via the macro now.